File: FrontEnd/src/app/Module/fornecedor.ts
import { Endereco } from "./endereco";
import { Produto } from "./produto";
import { TipoFornecedor } from "./TipoFornecedor ";

export class Fornecedor {
    nome: string = '';
    documento: string = '';
    tipoFornecedor: TipoFornecedor = TipoFornecedor.PessoaFisica;
    endereco: Endereco = new Endereco();
    ativo: boolean = false;
  
    // Relação com produtos
    produtos: Produto[] = [];
  
    constructor(init?: Partial<Fornecedor>) {
        Object.assign(this, init);
    }
  }

File: src/Dev.Domain/Models/Fornecedor.cs
﻿using Dev.Domain.Enum;
using DevIO.Business.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Dev.Domain.Models
{
    public class Fornecedor : Entity
    {
        public string Nome { get; set; }
        public string Documento { get; set; }
        public TipoFornecedor TipoFornecedor { get; set; }

        //public Endereco Endereco { get; set; }

        public bool Ativo { get; set; }

        /* EF Relations */
        public IEnumerable<Produto>? Produtos { get; set; }
    }
}
